app/controllers/viewing_party_controller.rb

```diff
@@ -0,0 +1,23 @@
+class ViewingPartyController < BaseController
+  def new
+    @movie = SearchResults.movie_details(params[:movie_id])
+  end
+
+  def create
+    party = Party.new(viewing_party_params)
+    create_party(party) if party.valid?
+  end
+
+  private
+
+  def create_party(event)
+    event.save
+    flash[:success] = "You've created a viewing party for #{params[:movie_title]}!"
+    redirect_to dashboard_index_path
+  end
+
+  def viewing_party_params
+    p = params.permit(:movie_title, :party_duration, :date, :time)
+    p.merge({ user_id: current_user.id })
```

Comment:
This line allows us to assign a viewing party to a user because a viewing party belongs to a user.